File: e-credit-backend/src/main/java/org/example/ecreditbackend/model/Demande.java
package org.example.ecreditbackend.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Entity
@Table(name = "demande")
@Data
@NoArgsConstructor
public class Demande {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_demande")

    private Integer idDemande;

    private String type;
    private String unite;
    private BigDecimal montant;

    private BigDecimal nbrEcheance;
    private String observations;
    private String etatDemande ;
    private Date dateCre=new Date();
    @OneToMany(cascade = CascadeType.PERSIST,mappedBy = "demande",orphanRemoval = true)

    private List<Garantie> listeGaranties=new ArrayList<>();
    @OneToOne(cascade = CascadeType.ALL)
    private Client client;
    @OneToOne(cascade = CascadeType.ALL)
    private Piece pieceJointe;

}

File: e-credit frontend/src/app/Models/Demande.ts
import { Client } from "./Client";
import { Garantie } from "./Garantie";
import { Piece } from "./Piece";

export interface Demande {
   
    idDemande: Number,
    dateCre:Date;
    etatDemande:String;

    client: Client,

    type: String,
    unite: String,
    montant: Number,
    nbrEcheance: Number,

    listeGaranties: Garantie[];
    
    pieceJointe: Piece;

    observations: String,
    

}